frappe/client.py

```diff
@@ -430,6 +430,9 @@ def validate_link(doctype: str, docname: str, fields=None):
 			frappe.PermissionError,
 		)
 
+	if frappe.get_meta(doctype).is_virtual:
+		return frappe._dict({"name": docname})
```

Comment:
Do frappe.get_doc to validate instead? 

It's not efficient, but virtual doctypes aren't meant to be.

Reply:
Makes sense, that was my first thought, but as you said inefficient so discarded. Will make the changes 👍